From 0fb21524bc5bf82d51c6cb097e4f10b1074b10a2 Mon Sep 17 00:00:00 2001
From: Steve Ellcey <sellcey@caviumnetworks.com>
Date: Wed, 15 Nov 2017 08:58:48 -0800
Subject: [PATCH 46/49] Check length of ifname before copying it into to ifreq
 structure.

          [BZ #22442]
          * sysdeps/unix/sysv/linux/if_index.c (__if_nametoindex):
          Check if ifname is too long.
---
 sysdeps/unix/sysv/linux/if_index.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sysdeps/unix/sysv/linux/if_index.c b/sysdeps/unix/sysv/linux/if_index.c
index 6bd6bda3b0..339078a47d 100644
--- a/sysdeps/unix/sysv/linux/if_index.c
+++ b/sysdeps/unix/sysv/linux/if_index.c
@@ -54,6 +54,12 @@ if_nametoindex (const char *ifname)
   if (fd < 0)
     return 0;
 
+  if (strlen (ifname) >= IFNAMSIZ)
+    {
+      __set_errno (ENODEV);
+      return 0;
+    }
+
   strncpy (ifr.ifr_name, ifname, sizeof (ifr.ifr_name));
   if (__ioctl (fd, SIOCGIFINDEX, &ifr) < 0)
     {
-- 
2.22.0

